tgui/packages/tgui/interfaces/DestinationTagger.tsx

```diff
@@ -7,6 +7,29 @@ type DestinationTaggerData = {
   currentTag: number;
 };
 
+/**
+ * Info about destinations that survives being re-ordered.
+ */
+type DestinationInfo = {
+  name: string;
+  sorting_id: number;
+};
+
+/**
+ * Sort destinations in alphabetical order,
+ * and wrap them in a way that preserves what ID to return.
+ * @param locations The raw, official list of destination tags.
+ * @returns The alphetically sorted list of destinations.
+ */
+const sortDestinations = (locations: string[]): DestinationInfo[] => {
+  return locations
+    .map((name, index) => ({
+      name: name.toUpperCase(),
+      sorting_id: index + 1,
+    }))
+    .sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
- localeCompare is extrelemy slow in Internet Explorer and should be avoided.
- You should have used a sortBy helper.

Reply:
That almost sounds crazy that it would be slower than using a script function that creates a second array but we're talking about IE so I believe it.